pkg/controller/core/workload_controller_test.go

```diff
@@ -407,6 +410,105 @@ func TestReconcile(t *testing.T) {
 		wantResult     reconcile.Result
 		reconcilerOpts []Option
 	}{
+		"reconcile DRA ResourceClaim -> quota accounted": {
+			workload: func() *kueue.Workload {
+				wl := utiltesting.MakeWorkload("wlWithDRAResourceClaim", "ns").
+					Queue("lq").
+					ReserveQuota(utiltesting.MakeAdmission("cq").Assignment("gpus", "flavor1", "1").Obj()).
+					PodSets(*utiltesting.MakePodSet(kueue.DefaultPodSetName, 1).Obj()).
+					Obj()
+				wl.Spec.PodSets[0].Template.Spec.ResourceClaims = []corev1.PodResourceClaim{{
+					Name: "gpu", ResourceClaimName: ptr.To("rc1"),
+				}}
+				wl.Spec.PodSets[0].Template.Spec.Containers = []corev1.Container{{
+					Name: "main", Image: "pause",
+					Resources: corev1.ResourceRequirements{Claims: []corev1.ResourceClaim{{Name: "gpu"}}},
+				}}
+				return wl
+			}(),
+			cq: utiltesting.MakeClusterQueue("cq").
+				ResourceGroup(
+					*utiltesting.MakeFlavorQuotas("flavor1").
+						Resource("gpus", "2").Obj(),
+				).Obj(),
+			lq:           utiltesting.MakeLocalQueue("lq", "ns").ClusterQueue("cq").Obj(),
+			wantWorkload: nil,
+			wantEvents: []utiltesting.EventRecord{
+				{
+					Key:       types.NamespacedName{Namespace: "ns", Name: "wlWithDRAResourceClaim"},
+					EventType: "Normal",
+					Reason:    "Admitted",
+					Message:   fmt.Sprintf("Admitted by ClusterQueue %s, wait time since reservation was %s", "cq", fakeClock.Since(metav1.NewTime(testStartTime).Time.Truncate(time.Second)).Truncate(time.Second)),
+				},
+			},
+		},
+		"reconcile DRA suspended workload (Active=false)": {
+			workload: func() *kueue.Workload {
+				wl := utiltesting.MakeWorkload("wlWithDRAResourceClaim", "ns").
+					Queue("lq").
+					// Admit first, but the workload is suspended
+					ReserveQuota(utiltesting.MakeAdmission("cq").Assignment("gpus", "flavor1", "1").Obj()).
+					Admitted(true).
+					Active(false).
+					PodSets(*utiltesting.MakePodSet(kueue.DefaultPodSetName, 1).Obj()).
+					Obj()
+				wl.Spec.PodSets[0].Template.Spec.ResourceClaims = []corev1.PodResourceClaim{{
+					Name: "gpu", ResourceClaimName: ptr.To("rc1"),
+				}}
+				wl.Spec.PodSets[0].Template.Spec.Containers = []corev1.Container{{
+					Name: "main", Image: "pause",
+					Resources: corev1.ResourceRequirements{Claims: []corev1.ResourceClaim{{Name: "gpu"}}},
+				}}
```

Comment:
It could be a follow up, but we should use the Wrapper (builder) concept. I think we should extend the PodSetWrapper to have a function accepting a REsourceClaim / ResourceClaimTemplate